apps/guardian/src/shared/store.ts

```diff
@@ -0,0 +1,133 @@
+import { atom } from 'jotai';
+import { atomEffect } from 'jotai-effect';
+import { AccountAddress, Timestamp } from '@concordium/web-sdk/types';
+import { ElectionManifest, ElectionParameters } from 'shared/types';
+import { getChecksumResource } from 'shared/util';
+
+import { GuardiansState, getElectionConfig, getGuardiansState } from './election-contract';
+import { WalletAccount, getAccounts } from './ffi';
+
+/**
+ * Representation of the election configration.
+ */
+export interface ElectionConfig {
+    /** The election start time */
+    start: Date;
+    /** The election end time */
+    end: Date;
+    /** The election description */
+    description: string;
+    /** The election manifest, used by election guard */
+    manifest: ElectionManifest;
+    /** The election parameters, used by election guard */
+    parameters: ElectionParameters;
+}
+
+/**
+ * Primitive atom for holding the {@linkcode ElectionConfig} of the election contract
+ */
+const electionConfigBaseAtom = atom<ElectionConfig | undefined>(undefined);
+
+/**
+ * Ensures an election config is fetched if the primitive atom holds no value.
+ */
+const ensureElectionConfigAtom = atomEffect((get, set) => {
+    if (get(electionConfigBaseAtom) !== undefined) {
+        return;
+    }
+
+    void getElectionConfig().then(async (config) => {
+        if (config === undefined) {
+            return undefined;
+        }
+
+        const electionManifestPromise = getChecksumResource<ElectionManifest>(config.election_manifest);
+        const electionParametersPromise = getChecksumResource<ElectionParameters>(config.election_parameters);
+
+        const [manifest, parameters] = await Promise.all([electionManifestPromise, electionParametersPromise]);
+
+        const mappedConfig: ElectionConfig = {
+            start: Timestamp.toDate(config.election_start),
+            end: Timestamp.toDate(config.election_end),
+            description: config.election_description,
+            manifest,
+            parameters,
+        };
+
+        set(electionConfigBaseAtom, mappedConfig);
+    });
+});
+
+/**
+ * Holds the configuration of the election contract. A reference to this should always be kept in the application root
+ * to avoid having to fetch the configuration more than once.
+ */
+export const electionConfigAtom = atom((get) => {
+    get(ensureElectionConfigAtom);
+    return get(electionConfigBaseAtom);
+});
+
+/** The interval at which the guardians state will refresh from the contract */
+const GUARDIANS_UPDATE_INTERVAL = 30000;
```

Comment:
Is it not reasonable? I want to avoid having to refresh manually and 30s is something we've used elsewhere (wallets) for refreshing state.

Do you have another proposal?

Reply:
It's not unreasonable :) 

So this is refreshing state from the chain right, looking for when other people will do stuff so we can proceed?

Reply:
Yes, basically it refreshed the data stored for all guardians to keep the UI up to date with which actions can be performed. At this point in the implementation, the only thing needed from the contract state however is the addresses of the guardians so I jumped the gun a little with the refreshing part of it. It will be used at a later point though, so will just keep it as is 😄